src/server/utils/settings-store/settings-store.ts

```diff
@@ -40,38 +40,36 @@ function readSettingsFactory(filepath: string, format: Format, inline = false) {
       .then(appSettingsJS => {
         if (inline) appSettingsJS = inlineVars(appSettingsJS, process.env);
         return AppSettings.fromJS(appSettingsJS, { visualizations: MANIFESTS });
-      })
-    );
+      });
   };
 }
 
 function writeSettingsFactory(filepath: string, format: Format) {
   return (appSettings: AppSettings) => {
-    return Q.fcall(() => {
-      switch (format) {
-        case "json":
-          return JSON.stringify(appSettings);
-        case "yaml":
-          return appSettingsToYAML(appSettings, false);
-        default:
-          throw new Error(`unsupported format '${format}'`);
-      }
-    })
-      .then(appSettingsYAML => {
-        return fs.writeFile(filepath, appSettingsYAML);
-      });
+    return noop()
```

Comment:
If I understand correctly, some branches in switch returns Promises and some not? I would prefer to call switch outside Promise and inside cases return Promise (like `Promise.resolve(JSON.stringify(...`). We would get rid of this one `then` and `noop`.